backend/src/controllers/datafile.controller.ts

```diff
@@ -38,96 +44,141 @@ export class DatafileController extends Controller {
+   * @throws NotFoundError if the document is not found.
    */
-  @Get("{fileId}")
+  @Get("{documentId}")
   @Response<NotFoundError>(404, "Not found")
   @SuccessResponse(200, "Datafile found.")
   public async getDatafile(
-    @Path() fileId: MongooseObjectId
+    @Path() documentId: MongooseObjectId
   ): Promise<Datafile> {
     this.setStatus(200);
-    return this.datafileService.get(fileId);
+    return this.datafileService.get(documentId);
   }
 
   /**
-   * Creates a file.
+   * Creates a document.
    *
-   * @param body - The data for creating the file.
+   * @param body - The data for creating the document.
    * @returns A promise that resolves to the created entity.
-   * @throws NotFoundError if the file is not found.
    */
-  @SuccessResponse(201, "Created successfully.") // Custom success response
+  @SuccessResponse(200, "Created successfully.") // Custom success response
   @Post()
   public async createDatafile(
     @Body() body: DatafileCreateParams
   ): Promise<Datafile> {
-    this.setStatus(201); // set return status 201
+    this.setStatus(200);
     return this.datafileService.create(body);
   }
 
   /**
-   * Deletes a file.
+   * Appends the uploaded file to a document with given ID.
    *
-   * @param fileId - The unique identifier of the file to delete.
+   * @param file - The file to append.
+   * @param documentID - The ID of the document to which to append the file
+   * @param fileType - Type of the uploaded file.
+   * @returns A promise that resolves to the updated entity.
+   * @throws OperationNotSupportedError if the file type is not supported.
+   * @throws FailedToParseError if there was an error in parsing the file.
+   * @throws WrongObjectTypeError selected document is not a NOTREFERENCED type.
+   * @throws NotFoundError if the entity is not found.
+   */
+  @SuccessResponse(200, "Appended file successfully.")
+  @Response<NotFoundError>(404, "Not found")
+  @Response<OperationNotSupportedError>(400, "File type not supported.")
+  @Response<FailedToParseError>(400, "Parsing failed.")
+  @Response<WrongObjectTypeError>(400, "Document is not NOTREFERENCED type.")
+  @Post("/file/append")
+  public async createDatafileFromFile(
+    @UploadedFile() file: Express.Multer.File,
+    @FormField() documentID: MongooseObjectId,
+    @FormField() fileType: SupportedRawFileTypes
+  ): Promise<Datafile> {
+    this.setStatus(200);
+    return this.datafileService.appendFile(file, documentID, fileType);
+  }
+
+  /**
+   * Creates all datafiles from the uploaded dataset file
+   *
+   * @param file - The file to append.
+   * @param dataset - Type of the dataset provided.
+   * @returns A promise that resolves to all created entities.
+   * @throws OperationNotSupportedError if the dataset type is not supported.
+   */
+  @SuccessResponse(200, "Appended file successfully.")
+  @Response<OperationNotSupportedError>(400, "Dataset not supported.")
+  @Post("/file/dataset")
```

Comment:
i find the route "datafiles/file/dataset" confusion to be honest
```suggestion
  @Post("/fromFile")
```